google/cloud/bigtable/batcher.py

```diff
@@ -128,8 +129,9 @@ def control_flow(self, batch_info):
         Calculate the resources used by this batch
         """
 
-        self.inflight_mutations += batch_info.mutations_count
-        self.inflight_size += batch_info.mutations_size
+        with self._lock:
+            self.inflight_mutations += batch_info.mutations_count
+            self.inflight_size += batch_info.mutations_size
         self.set_flow_control_status()
```

Comment:
I believe we can keep set_flow_control_status out of the lock. Even if it occasionally gets clobbered, every `add` and `release` ends with another call to set_flow_control_status, so it should be eventually correct.